code/worker.py

```diff
@@ -1,38 +1,40 @@
 #!/usr/bin/env python3
+import argparse
+
+import requests
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'appimage_url', metavar='URL',
+        help="URL of the AppImage file to query")
+    ns = parser.parse_args()
+    r = requests.get(ns.appimage_url, stream=True)
+    r.raise_for_status()
+    header = r.raw.read(8+3)
+    tag = header[8:8+3]
+    name = r.url
+    if tag == b'AI\x01':
+        print("{} looks like a valid type-1 AppImage file".format(name))
+    elif tag == b'AI\x02':
+        print("{} looks like a valid type-2 AppImage file".format(name))
+    else:
+        raise SystemExit("{} doesn't look like a valid appimage file".format(name))
+    # If 0x414902 then extract desktop file like so:
```

Comment:
I kept them because I didn't touch this part of the logic but I can yank them if you feel strongly about it.